MaxText/max_utils.py

```diff
@@ -216,6 +217,7 @@ def maybe_initialize_jax_distributed_system(raw_keys):
   ) or raw_keys["hardware"] == "gpu_multiprocess":
     max_logging.log("Attempting to initialize the jax distributed system...")
     jax.distributed.initialize()
+    ocp.multihost.utils.initialize_runtime_to_distributed_ids()
```

Comment:
Don't you also need https://github.com/google/orbax/blob/main/checkpoint/orbax/checkpoint/multihost/utils.py#L37 flag set? I don't see that anywhere.

Reply:
It is not needed for saving. Even for restoring, I am not sure it's required since the runtime-to-distributed id mapping is saved, and initialize_runtime_to_distributed_ids() doesn't depend on that flag. In fact, my testing without that flag didn't reveal any issues.

Reply:
I submitted https://github.com/google/maxtext/pull/724 to use the latest Orbax (0.5.20) and JAX (0.4.30) and made sure that these versions were indeed being used in the actual run.

As expected, without setting the flag, no logs in https://github.com/google/orbax/blob/4cae60f87b1abfe09bfbdfea03718d4f93d5b755/checkpoint/orbax/checkpoint/multihost/utils.py#L251 are present, so `jax.process_index()` is used as multihost `process_index()`.

Still, saving never fails in 10+ tries. At this point, I suggest that we revisit the Orbax code carefully and understand this behavior before moving forward. With the many hacks, it's also a good opportunity for us to have a careful review of the e2e code path.

Reply:
To summarize the internal investigation: setting this flag is indeed not necessary for saving. Will see its relevance in the restore piece.